classes/Link.php

```diff
@@ -428,6 +428,10 @@ public function getCategoryLink(
         $idShop = null,
         $relativeProtocol = false
     ) {
+        if ($category === 0) {
```

Comment:
But you could probably do even better than this, because as you can see a few line below the `$category` can actually be an Object an array or an int

So you probably should add your deprecation notice on L453 by checking the `$params['id']` value as well Same suggestion if you do that, you'd better cast it as an int if you use the strict equality because even in legacy objects the field are not always correctly typed (event the ID)

Reply:
Also as discussed live: maybe we can test this behavior.

Unfortunately `classes/Link.php` is a legacy class strongly coupled to multiple other legacy classes so unit tests cannot be done. But we have an integration test with some mocking stuff here https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Classes/LinkTest.php maybe we can expand this Test to be able to test your new case

Reply:
I'm gonna need help to add that test case 😄

Reply:
I checked Link class, there is so much coupling to singletons 😭 let's forget about it